ResearchKit/Common/ORKLocationSelectionView.m

```diff
@@ -232,27 +234,28 @@ - (void)reverseGeocodeAndDisplay:(CLLocation *)location {
             [strongSelf setAnswer:ORKNullAnswerValue()];
         } else {
             CLPlacemark *placemark = [placemarks lastObject];
-            [strongSelf setAnswer:placemark];
+            [strongSelf setAnswer:[[ORKLocation alloc] initWithPlaceMark:placemark]];
         }
     }];
 }
 
-- (void)setAnswer:(id)answer {
+- (void)setAnswer:(ORKLocation *)answer {
     
     [_mapView removeAnnotations:_mapView.annotations];
     
-    if ([[answer class] isSubclassOfClass:[CLPlacemark class]]) {
-        _answer = [[ORKPlacemark alloc] initWithPlacemark:answer];
-    } else {
+    if (NO == [[answer class] isSubclassOfClass:[ORKLocation class]]) {
         _answer = answer == ORKNullAnswerValue() ? ORKNullAnswerValue() : nil;
+    } else {
+        _answer = answer;
     }
```

Comment:
Or maybe change this whole if else into:
`_answer = ([[answer class] isSubclassOfClass:[ORKLocation class]]) ? answer : ORKNullAnswerValue()` ?

Do we need to store nil and ORKNullAnswerValue for some purpose?